### PR TITLE
Add backend

### LINKED ISSUE #18: Database
Create a database most likely using Turso to store user information.  Clerk will store login information including passwords so the database will not store this. The database's primary key is the user's Clerk ID.

### PR DESCRIPTION
**Pull request by jsun969**
### Description
Add backend

### Changes Made
1. Add database integration with drizzle orm and turso.
2. Add api for member join.

### Related Issues
close #18 

### Additional Notes
Read the updated readme and env example to initialize database first.

### CODE REVIEW

File: src/constants/student-info.ts
Comment (rayokamoto) [inline src/constants/student-info.ts:1]: Could we maybe have a newline to separate out each of the constants?

File: src/db/schema.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    studentStatus: text('student_status', { enum: STUDENT_STATUSES }).notNull(),
+    studentId: text('student_id'),
+    gender: text('gender', { enum: GENDERS }).notNull(),
+    ageBracket: text('age_bracket', { enum: AGE_BRACKETS }).notNull(),
+    degree: text('degree', { enum: DEGREES }),
+    studentType: text('student_type', { enum: STUDENT_TYPES }),
+
+    emailPreferences: text('email_preferences', { mode: 'json' }),
+
+    membershipExpiredAt: integer('membership_expired_at', { mode: 'timestamp' }),
```
Comment (rayokamoto) [inline src/db/schema.ts:31]: Change to `membershipExpiresAt` and `membership_expires_at` as membership could still be ongoing (it doesn't really make sense to say "expired" if it hasn't, if you get what I mean).

**Review by rayokamoto — COMMENTED** [review]
Looking good, only a few minor changes needed.

File: src/db/schema.ts (hunk 1 above)
Comment (jsun969) [reply src/db/schema.ts:31]: I really need to learn more English 🤣

**Review by rayokamoto — APPROVED** [review]

**Pull request merged**